wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/UnsupportedEntityIdValueImpl.java

```diff
@@ -0,0 +1,160 @@
+package org.wikidata.wdtk.datamodel.implementation;
+
+/*-
+ * #%L
+ * Wikidata Toolkit Data Model
+ * %%
+ * Copyright (C) 2014 - 2019 Wikidata Toolkit Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.wikidata.wdtk.datamodel.helpers.Equality;
+import org.wikidata.wdtk.datamodel.helpers.Hash;
+import org.wikidata.wdtk.datamodel.helpers.ToString;
+import org.wikidata.wdtk.datamodel.interfaces.UnsupportedEntityIdValue;
+import org.wikidata.wdtk.datamodel.interfaces.ValueVisitor;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonDeserializer.None;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * Represents a entity id value of an unsupported type.
+ * We can still "deserialize" it by just storing its
+ * JSON representation, so that it can be serialized
+ * back to its original representation.
+ * This avoids parsing failures on documents containing
+ * these values.
+ * 
+ * @author Antonin Delpeuch
+ */
+@JsonDeserialize(using = None.class)
+public class UnsupportedEntityIdValueImpl extends ValueImpl implements UnsupportedEntityIdValue {
+	
+	private final JacksonIdValue value;
+	private final String siteIri;
+
+	@JsonCreator
+	private UnsupportedEntityIdValueImpl(
+			@JsonProperty("value")
+			JacksonIdValue value,
+			@JacksonInject("siteIri")
+			String siteIri) {
+		super(JSON_VALUE_TYPE_ENTITY_ID);
+		this.value = value;
+		this.siteIri = siteIri;
+	}
+
+	private static class JacksonIdValue {
+		private final String entityType;
+		private final String id;
+		private final Map<String, JsonNode> contents;
+		
+		@JsonCreator
+		private JacksonIdValue(
+				@JsonProperty("entity-type")
+				String entityType,
```

Comment:
I believe that now "id" is the only key that is always supposed to exist. We should make sure we do not fail in this case.

Reply:
Makes sense. Do you know in what scenario the `entity-type` can go missing?

Reply:
I believed it was the case for of Lexemes, but it seems it's not the case anymore. Example: https://www.wikidata.org/wiki/Special:EntityData/L2.json